backends/platform/sdl/macosx/appmenu_osx.mm

```diff
@@ -85,14 +85,27 @@ static void openFromBundle(NSString *file) {
 		}
 	}
 
+	// RTF and HTML files are widely recognized and we can rely on the default
+	// file association working for those. For the other ones this might not be
+	// the case so we explicitly indicate they should be open with TextEdit.
 	if (path) {
-		// RTF and HTML files are widely recognized and we can rely on the default
-		// file association working for those. For the other ones this might not be
-		// the case so we explicitly indicate they should be open with TextEdit.
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_15
```

Comment:
I will need to double check as I have always been a bit confused by these two defines, but I think we need to use `__MAC_OS_X_VERSION_MIN_REQUIRED` here and not `__MAC_OS_X_VERSION_MAX_ALLOWED`.

From what I remember the latter corresponds to the SDK used for the compilation (and we are indeed not allowed to use APIs from more recent SDKs) while the former corresponds to the version we are targeting (typically for the official intel 64 bits release I target Mac OS X 10.9 but compile with a more recent SDK), so while the compilation might work when using 10.5 APIs, it will likely not run when trying to run on older systems.

Reply:
Thanks They often confuse me too, and I really needed something as clear as your synthesis!

I did some builds targeting macOS 10.14, 10.15, 11.0 or 12.0 (with the 12.0 or 13.0 SDK) in Xcode. With `MAC_OS_X_VERSION_MIN_REQUIRED`, the embedded symbols now look more correct (i.e. no `openUrls` when targeting 10.14).